Doc/library/stdtypes.rst

```diff
@@ -3158,6 +3158,97 @@ objects.
 
       .. versionadded:: 3.14
 
+   .. method:: take_bytes(n=None, /)
+
+      Take the first *n* bytes as an immutable :class:`bytes`. Defaults to all
+      bytes.
+
+      If *n* is negative indexes from the end and takes the first :func:`len`
+      minus *n* bytes. If *n* is out of bounds raises :exc:`IndexError`.
+
+      Taking less than the full length will leave remaining bytes in the
+      :class:`bytearray` which requires a copy. If the remaining bytes should be
+      discarded use :func:`~bytearray.resize` or :keyword:`del` to truncate
+      then :func:`~bytearray.take_bytes` without a size.
+
+      .. impl-detail::
+
+         CPython implements this as a zero-copy operation making it a very
+         efficient way to make a :class:`bytes` from a :class:`bytearray`.
```

Comment:
Only when n is None, no?

Reply:
Yea, I've been struggling to find a concise wording "minimal copying" leaves a lot more wiggle room to me; can just add the caveat "when taking all bytes" (`ba.take_bytes(len(ba))` also doesn't copy)